### PR TITLE
[Exam] several style fixes & tooltips

### PR DESCRIPTION
**Pull request by filip-gregurevic**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.

### Motivation and Context

The save & continue / save button did not make sense and hiding it by `ngIf` caused problems with the layout of the nav.
On certain screen sizes the buttons in the exam navigation are overlapping. See https://github.com/ls1intum/Artemis/pull/1767#issuecomment-651818691

### Description

The save & continue / save button is now set to not visible in order to keep the layout.
The indicator of the exercise now switches earlier to the smaller display to prevent overlapping. (No overlap is only guaranteed for screen at least as wide as iPads (768px) but it also works until approx. 500px)
Added tooltips for the buttons.

### Steps for Testing

1. Log in to Artemis
2. Prepare an exam with at least one programming exercise
4. Check that the save & continue / save button is not shown when working on the programming exercise
5. Make sure the navigation is displayed correctly even when changing the window size

### CODE REVIEW

**Comment by filip-gregurevic** [comment]
> I've tested this on TS1 and this is the most overlap I've been able to generate:
> 
> With a text exercise:
> image
> 
> With a programming exercise (ignore the highlighting on time):
> 
> image
> 
> It also seems that once I get to that point, the gray button to advance to the next exercise stops being clickable.

@pzdr7 Thanks for testing. Could you also provide the screen size at which it overlaps? So that I can replicate and fix it.

**Comment by pzdr7** [comment]
Fixed, I've added the pixel measurements to the screenshots (as provided by chrome's dev tools).

File: src/main/webapp/i18n/de/exam.json [hunk 1]
```diff
@@ -49,7 +49,11 @@
             "submitOtherExercise": "Speichern & Weiter",
             "submitLastExercise": "Speichern",
             "greeting": "Hallo {{ fullName }}, willkommen zu \"{{ title }}\"",
-            "points": "Punkte"
+            "points": "Punkte",
+            "syncedAndSubmitted": "Aufgabe ist mit dem Server synchronisiert und eingereicht",
+            "syncedAndNotSubmitted": "Aufgabe ist mit dem Server synchronisiert aber nicht eingereicht",
+            "notSyncedAndSubmitted": "Aufgabe ist nicht mit dem Server synchronisiert aber eingereicht",
+            "notSyncedAndNotSubmitted": "ufgabe ist nicht mit dem Server synchronisiert und nicht eingereicht"
```
Comment (krusche) [inline src/main/webapp/i18n/de/exam.json:56]:
```suggestion
            "notSyncedAndNotSubmitted": "Aufgabe ist nicht mit dem Server synchronisiert und nicht eingereicht"
```

**Comment by krusche** [comment]
We have a merge conflict now :-(
And I don't think the submitted state is interesting for the tooltips.
I think only the difference between blue (synced) and yellow (not synced) is important.

Only for programming exercises, we should differentiate that yellow can mean the they have unsaved or uncommitted changes

**Comment by filip-gregurevic** [comment]
> We have a merge conflict now :-(
> And I don't think the submitted state is interesting for the tooltips.
> I think only the difference between blue (synced) and yellow (not synced) is important.
> 
> Only for programming exercises, we should differentiate that yellow can mean the they have unsaved or uncommitted changes

I fixed the conflicts and the longer tooltips including the submitted state are now only shown for programming exercises

**Comment by jpbernius** [comment]
image
I am unsure of the tooltips. How is this not submitted?

**Comment by jpbernius** [comment]
Still, in terms of programming exercise: How do I submit it? The submit button does not do it in terms of tooltips!

**Comment by krusche** [comment]
I fixed it

**Pull request merged**